### PR TITLE
[WIP] Added some math functions and helpers

### PR DESCRIPTION
**Pull request by briederer**
Switched the order of upper and lower error for datapoints because it seems more reasonable to use it in this order `(lower,upper)`.

Added some default constructors and type-specific ones which are needed when calling some functions with `DataPoint` types as an argument.

Need an opinion by @fzierler on how we treat the systematic uncertainties for things like relative error and region of datapoints.

### CODE REVIEW

File: src/datapoint.jl [hunk 1]
```diff
@@ -34,14 +34,19 @@ Returns the systematic uncertainty of the data point as a tuple of upper and
 lower bounds. If none is provided it will default to zero.
 """
 syserr(x::DataPoint) = x.sys_err
+
+relstaterr(x::DataPoint) = x.stat_err / abs(x.val)
+
+statbounds(x::DataPoint) = x.val .+ (-1,1).*x.stat_err
```
Comment (fzierler) [inline src/datapoint.jl]: We could create the tuple in the end instead of using broadcasting.
`statbounds(x::DataPoint) = (x.val - x.stat_err,x.val + x.stat_err)`

File: src/datapoint.jl [hunk 2]
```diff
@@ -66,16 +71,21 @@ const ±(val, err::NamedTuple) = DataPoint(val, hasproperty(err,:stat) ? err.sta
...
 ## Pretty-printing
 # Printing of results
 function show(io::IO, ::MIME"text/plain", D::DataPoint{T}; drop::Bool = true) where {T}
     println(io, "DataPoint{",T,"}:")
-    print(io, "Value → ", D.val)
-    if (!drop || D.stat_err != (0,0)) print(io, "\tStatistic unc. → ", (1,-1) .* D.stat_err); end
-    if (!drop || D.sys_err != (0,0)) print(io, "\tSystematic unc. → ", (1,-1) .* D.sys_err); end
+    print(io, "Value → ")
+    show(io, D.val)
+    if (!drop || D.stat_err != (0,0)) print(io, "\tStatistic unc. → ", (-1,1) .* D.stat_err); end
```
Comment (fzierler) [inline src/datapoint.jl]: Maybe printing them as `val +- stat +- sys` is more space efficient and easier to read? But that can go in a separate PR.

**Review by fzierler — COMMENTED** [review]
Looks good to me.

**Comment by fzierler** [comment]
> Need an opinion by @fzierler on how we treat the systematic uncertainties for things like relative error and region of datapoints.

I don't understand what you mean by that.

File: src/datapoint.jl (hunk 1 above)
Comment (briederer) [reply src/datapoint.jl]: In general `x.stat_err` is a tuple and therefore using the broadcasting. Otherwise your suggestion would need to look like 
` statbounds(x::DataPoint) = (x.val - x.stat_err[1],x.val + x.stat_err[2])` and I don't think that this is a good idea.

File: src/datapoint.jl (hunk 2 above)
Comment (briederer) [reply src/datapoint.jl]: Yes indeed I am not happy with the printing at this moment, but this is something for another PR.
Actually when the `io::IO` is `:compact` (e.g. in arrays) then it already prints your proposed way.

**Comment by briederer** [comment]
> > Need an opinion by @fzierler on how we treat the systematic uncertainties for things like relative error and region of datapoints.
> 
> I don't understand what you mean by that.

I mean if we want to have something like `relerror(D::DataPoint)` how is the correct way to handle systematic uncertainties?
Should we then have one tuple with `(stat_err .+ sys_err)./val` or return two tuples for the errors independently?

I am also not sure if we even should keep the systematic error after all.

File: src/datapoint.jl (hunk 1 above)
Comment (fzierler) [reply src/datapoint.jl]: I see. Then I'm fine with having it this way. We could use `@.` but that is up to your preference.

**Comment by fzierler** [comment]
> I am also not sure if we even should keep the systematic error after all.

I agree. Let's get rid of them for now

**Comment by briederer** [comment]
Moved to #25
